### ISSUE TITLE
[rllib] Raylet crash with slow tasks [Failed to wait for objects]

### ISSUE THREAD

**Issue by ericl** [ray-project/ray#3813]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04
- **Ray installed from (source or binary)**: binary
- **Ray version**: 0.6.2 (latest)
- **Python version**: 2.7
- **Exact command to reproduce**:
```
from ray.tune import run_experiments
import ray

ray.init(redis_address="localhost:6379")

run_experiments({
  "crash": {
    "run": "PPO",
    "num_samples": 40,
    "env": "CartPole-v0",
    "config": {
      "num_workers": 7,
      "sample_batch_size": 10000,
    },
  }
})
```

### Describe the problem
On a multi-node Ray cluster, this crashes on startup with 
```
WARNING: Logging before InitGoogleLogging() is written to STDERR
F0120 23:37:09.576818 116971 raylet_extension.cc:194] 
Check failed: _s.ok() [RayletClient] Failed to wait for objects.:
IOError: [RayletClient] Raylet connection closed.
```

You might have to run it a couple times to reproduce, but it's fairly easy to.

**Comment by ericl** [comment]
We don't dump core for fatal errors right? I have the stack:
```
I0121 00:04:29.623526  1552 node_manager.cc:418] Actor 2f90f3e7d536f64e98be7edda3b67567d4375669 is disconnected, because its node 8144fff2cf16224e7137078949e99a4703b50fbf is removed from cluster. It may be reconstructed.
I0121 00:04:29.623596  1552 node_manager.cc:1851] Failed to forward task 000000006e0c08d7d76eaefdfb6fe2363a2c9c30 to node manager d817830f3a2a0298c74f73b019b62b2ae0006a02
I0121 00:04:29.623693  1552 node_manager.cc:1851] Failed to forward task 00000000af68abac23d29c6d4fe6ab0634040961 to node manager 8144fff2cf16224e7137078949e99a4703b50fbf
F0121 00:04:29.623759  1552 lineage_cache.cc:338]  Check failed: entry 
*** Check failure stack trace: ***
*** Aborted at 1548029069 (unix time) try "date -d @1548029069" if you are using GNU date ***
PC: @                0x0 (unknown)
*** SIGABRT (@0x3e800000610) received by PID 1552 (TID 0x7fd4dfee0740) from PID 1552; stack trace: ***
    @     0x7fd4dfaca390 (unknown)
    @     0x7fd4dec7b428 gsignal
    @     0x7fd4dec7d02a abort
    @           0x5b6b0d google::logging_fail()
    @           0x5b6b36 google::LogMessage::Fail()
    @           0x5b6a82 google::LogMessage::SendToLog()
    @           0x5b6406 google::LogMessage::Flush()
    @           0x5b6215 google::LogMessage::~LogMessage()
    @           0x502e18 ray::RayLog::~RayLog()
    @           0x54f9f2 ray::raylet::LineageCache::GetUncommittedLineage()
    @           0x5388fe ray::raylet::NodeManager::ForwardTask()
    @           0x5390b5 ray::raylet::NodeManager::ForwardTaskOrResubmit()
    @           0x53b2b4 ray::raylet::NodeManager::ScheduleTasks()
    @           0x53f322 _ZZN3ray6raylet11NodeManager21ForwardTaskOrResubmitERKNS0_4TaskERKNS_8UniqueIDEENKUlNS_6StatusEE_clES8_.isra.1058
    @           0x53f465 _ZNSt17_Function_handlerIFvRKN3ray6StatusEEZNS0_6raylet11NodeManager21ForwardTaskOrResubmitERKNS5_4TaskERKNS0_8UniqueIDEEUlS1_E_E9_M_invokeERKSt9_Any_dataS3_
    @           0x540ab7 _ZNSt17_Function_handlerIFvRKN3ray6StatusEEZNS0_6raylet11NodeManager11ForwardTaskERKNS5_4TaskERKNS0_8UniqueIDERKSt8functionIS4_EEUlS1_E_E9_M_invokeERKSt9_Any_dataS3_
    @           0x50cc7f _ZZN3ray16ServerConnectionIN5boost4asio2ip3tcpEE13DoAsyncWritesEvENKUlRKNS1_6system10error_codeEmE_clES9_m
    @           0x50d0b4 _ZN5boost4asio6detail8write_opINS0_19basic_stream_socketINS0_2ip3tcpEEESt6vectorINS0_12const_bufferESaIS8_EEN9__gnu_cxx17__normal_iteratorIPKS8_SA_EENS1_14transfer_all_tEZN3ray16ServerConnectionIS5_E13DoAsyncWritesEvEUlRKNS_6system10error_codeEmE_EclESN_mi
    @           0x50d33b _ZN5boost4asio6detail23reactive_socket_send_opINS1_16prepared_buffersINS0_12const_bufferELm64EEENS1_8write_opINS0_19basic_stream_socketINS0_2ip3tcpEEESt6vectorIS4_SaIS4_EEN9__gnu_cxx17__normal_iteratorIPKS4_SD_EENS1_14transfer_all_tEZN3ray16ServerConnectionIS9_E13DoAsyncWritesEvEUlRKNS_6system10error_codeEmE_EEE11do_completeEPvPNS1_19scheduler_operationESQ_m 
    @           0x4bce0d boost::asio::detail::scheduler::run()
    @           0x4b378a main
    @     0x7fd4dec66830 __libc_start_main
    @           0x4b8ed1 (unknown)
```
debug state:
```
NodeManager:
InitialConfigResources: {CPU,64.000000},{GPU,4.000000}
ClusterResources:
d817830f3a2a0298c74f73b019b62b2ae0006a02: 
- total: {GPU,0.000000},{CPU,64.000000}
- avail: {GPU,0.000000},{CPU,30.000000}
b4395ba042ff9ac1ba583e60112ecf3bc516f3ca: 
- total: {GPU,0.000000},{CPU,64.000000}
- avail: {GPU,0.000000},{CPU,0.000000}
038acb1b26fc74684c2dafade504971bcf23db6a: 
- total: {GPU,0.000000},{CPU,64.000000}
- avail: {GPU,0.000000},{CPU,29.000000}
8144fff2cf16224e7137078949e99a4703b50fbf: 
- total: {GPU,0.000000},{CPU,64.000000}
- avail: {GPU,0.000000},{CPU,34.000000}
32ad4a8853ff2cbb886c84e90b4620615a72789f: 
- total: {CPU,64.000000},{GPU,4.000000}
- avail: {CPU,0.000000},{GPU,4.000000}
ObjectManager:
- num local objects: 2213
- num active wait requests: 12
- num unfulfilled push requests: 0
- num pull requests: 68
- num buffered profile events: 0
ObjectDirectory:
- num listeners: 128
- num eviction entries: 2213
ObjectStoreNotificationManager:
- num adds processed: 2213
- num removes processed: 0
BufferPool:
- get buffer state map size: 0
- create buffer state map size: 0
ConnectionPool:
- num message send connections: 4
- num transfer send connections: 4
- num avail message send connections: 4
- num avail transfer send connections: 4
- num message receive connections: 4
- num transfer receive connections: 4
AsyncGcsClient:
- TaskTable: num lookups: 69, num adds: 762
- ActorTable: num lookups: 204, num appends: 192
- TaskReconstructionLog: num lookups: 0, num appends: 69
- TaskLeaseTable: num lookups: 0, num adds: 1410
- HeartbeatTable: num lookups: 0, num adds: 2301
- ErrorTable: num lookups: 0, num appends: 0
- ProfileTable: num lookups: 0, num appends: 1254
- ClientTable: num lookups: 0, num appends: 1, cache size: 7, num removed: 2
- DriverTable: num lookups: 0, num appends: 1
WorkerPool:
- num workers: 74
- num drivers: 1
SchedulingQueue:
- num placeable tasks: 0
- num waiting tasks: 0
- num ready tasks: 1
- num running tasks: 37
- num infeasible tasks: 0
- num methods waiting for actor creation: 79
ReconstructionPolicy:
- num reconstructing: 68
TaskDependencyManager:
- task dep map size: 92
- task req map size: 131
- req objects map size: 68
- local objects map size: 2277
- pending tasks map size: 160
LineageCache:
- committed tasks: 0
- child map size: 96
- num subscribed tasks: 5
- lineage size: 113
ActorRegistry:
- num live actors: 222
- num reconstructing actors: 222
- num dead actors: 221
- max num handles: 1
RemoteConnections:
d817830f3a2a0298c74f73b019b62b2ae0006a02: 
- bytes read: 0
- bytes written: 265352
- num async writes: 228
- num sync writes: 0
- writing: 1
- pending async bytes: 71280
b4395ba042ff9ac1ba583e60112ecf3bc516f3ca: 
- bytes read: 0
- bytes written: 425512
- num async writes: 297
- num sync writes: 0
- writing: 0
- pending async bytes: 0
038acb1b26fc74684c2dafade504971bcf23db6a: 
- bytes read: 0
- bytes written: 249960
- num async writes: 224
- num sync writes: 0
- writing: 1
- pending async bytes: 60944
8144fff2cf16224e7137078949e99a4703b50fbf: 
- bytes read: 0
- bytes written: 234848
- num async writes: 188
- num sync writes: 0
- writing: 1
- pending async bytes: 91120
```

**Comment by ericl** [comment]
The error is here: https://github.com/ray-project/ray/blob/bfcf254e528b76b631bbfc24c9a8dcf47dd2ddab/src/ray/raylet/lineage_cache.cc#L338

It seems that when forwarding the actor task fails in a certain way, you hit this CHECK in lineage cache. What's a better way of handling this failure? Should we return an empty lineage object and log an error instead?

**Comment by stephanie-wang** [comment]
I believe this is closed by #3681 and #3860.
